### PR TITLE
[workflow] Change `step` to `task` in workflow.

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?
Recently we deprecate step in workflow. This PR wrap up everything and replace step to task in workflow to reflect the recent changes.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/workflow/api.py [hunk 1]
```diff
@@ -299,19 +299,19 @@ def get_output(workflow_id: str, *, name: Optional[str] = None) -> Any:
...
         workflow_id: The workflow to get the output of.
-        name: If set, fetch the specific step instead of the output of the
+        name: If set, fetch the specific task instead of the output of the
             workflow.
 
     Examples:
         >>> from ray import workflow
         >>> start_trip = ... # doctest: +SKIP
-        >>> trip = start_trip.options(name="trip").step() # doctest: +SKIP
+        >>> trip = start_trip.options(name="trip").task() # doctest: +SKIP
```
Comment (suquark) [inline python/ray/workflow/api.py:308]: `bind()`. Same in other places.

File: python/ray/workflow/api.py [hunk 2]
```diff
@@ -299,19 +299,19 @@ def get_output(workflow_id: str, *, name: Optional[str] = None) -> Any:
...
         >>> res1 = trip.run_async(workflow_id="trip1") # doctest: +SKIP
```
Comment (suquark) [inline python/ray/workflow/api.py:309]: we should use `workflow.run_async`. Same in other places

**Review by suquark — CHANGES_REQUESTED** [review]

File: python/ray/workflow/api.py [hunk 3]
```diff
@@ -264,8 +264,8 @@ def resume_async(workflow_id: str) -> ray.ObjectRef:
     Examples:
         >>> from ray import workflow
         >>> start_trip = ... # doctest: +SKIP
-        >>> trip = start_trip.step() # doctest: +SKIP
-        >>> res1 = trip.run_async(workflow_id="trip1") # doctest: +SKIP
+        >>> trip = start_trip.bind() # doctest: +SKIP
+        >>> res1 = workflow.run_async(trip, workflow_id="trip1") # doctest: +SKIP
         >>> res2 = workflow.resume("trip1") # doctest: +SKIP
```
Comment (suquark) [inline python/ray/workflow/api.py:269]: now it is `resume_async`, maybe also in other places

File: python/ray/workflow/api.py [hunk 4]
```diff
@@ -598,56 +599,56 @@ def get_metadata(workflow_id: str, name: Optional[str] = None) -> Dict[str, Any]
...
             the metadata of the workflow.
 
     Examples:
         >>> from ray import workflow
         >>> trip = ... # doctest: +SKIP
-        >>> workflow_step = trip.options( # doctest: +SKIP
-        ...     name="trip", metadata={"k1": "v1"}).step()
-        >>> workflow_step.run( # doctest: +SKIP
+        >>> workflow_task = trip.options( # doctest: +SKIP
+        ...     name="trip", metadata={"k1": "v1"}).bind()
```
Comment (suquark) [inline python/ray/workflow/api.py:626]: `**workflow.options(name=..., ....)`. Same in other places

File: python/ray/workflow/api.py (hunk 3 above)
Comment (iycheng) [reply python/ray/workflow/api.py:269]: I see. Thanks for catching!

**Comment by iycheng** [comment]
(push for DCO)

**Review by suquark — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**